src/test/crimson/test_messenger_recovery.cc

```diff
@@ -0,0 +1,801 @@
+      }).then([&test_msg] {
+      return seastar::do_until([&test_msg] {
+        return test_msg.clients_count() == test_msg.clients; },
+      [&test_msg] {
+        return test_msg.init_client(entity_name_t::CLIENT(-1),
+                             "client", get_nonce());
+	});
+      }).then([&test_msg] {
+        return test_msg.generate_connections(50);
+      }).then([&test_msg] {
+        return test_msg.random_op(100);
+      }).then([&test_msg] {
+	return seastar::with_timeout(
+	  seastar::lowres_clock::now() + 300s,
+	  test_msg.wait_for_done());
+      }).then([] {
+        logger().info("test_stress_1() DONE");
+      }).handle_exception([] (auto eptr) {
+        logger().error(
+          "test_stress_1() failed: got exception {}",
+          eptr);
+        throw;
+      });
+    });
+}
+
+// Test Parameters:
+// Clients: 16             (stateful)
+// Servers: 32             (loseless)
+// Connections: 100        (Generated between random clients/server)
+// Random Operations: 1000 (Generate/Drop Connection, Send Message, Sleep)
+seastar::future<> test_stress_2()
+{
+
+  logger().info("test_stress_2():");
+
+  SyntheticWorkload test_msg(16, 32, 100,
+                             crimson::net::SocketPolicy::stateful_server(0),
+                             crimson::net::SocketPolicy::lossless_client(0));
+
+  return seastar::do_with(test_msg, []
+    (SyntheticWorkload& test_msg) {
+      return seastar::do_until([&test_msg] {
+        return test_msg.servers_count() == test_msg.servers; },
+      [&test_msg] {
+        entity_addr_t bind_addr = get_server_addr();
+        bind_addr.set_type(entity_addr_t::TYPE_MSGR2);
+	uint64_t server_num = get_nonce();
+        return test_msg.init_server(entity_name_t::OSD(server_num),
+                             "server", server_num , bind_addr);
+      }).then([&test_msg] {
+      return seastar::do_until([&test_msg] {
+        return test_msg.clients_count() == test_msg.clients; },
+      [&test_msg] {
+        return test_msg.init_client(entity_name_t::CLIENT(-1),
+                             "client", get_nonce());
+	});
+      }).then([&test_msg] {
+        return test_msg.generate_connections(50);
+      }).then([&test_msg] {
+        return test_msg.random_op(100);
+      }).then([&test_msg] {
+	return seastar::with_timeout(
+	  seastar::lowres_clock::now() + 300s,
+	  test_msg.wait_for_done());
+      }).then([] {
+        logger().info("test_stress_2() DONE");
+      }).handle_exception([] (auto eptr) {
+        logger().error(
+          "test_stress_2() failed: got exception {}",
+          eptr);
+        throw;
+      });
+    });
+}
+
+// Test Parameters:
+// Clients: 8              (loseless)
+// Servers: 32             (statefull)
+// Connections: 100        (Generated between random clients/server)
+// Random Operations: 1000 (Generate/Drop Connection, Send Message, Sleep)
+seastar::future<> test_injection_1()
```

Comment:
Seems these test mostly share the same logic. It would be good to introduce a struct
```
struct thrash_params {
  std::size_t clients;
  std::size_t servers;
  ... connections, random operations, failure ratio, delay ...
};
```
And feed it to the general test function.